src/API/src/db/occurrence/occurrence.resolver.ts

```diff
@@ -110,14 +125,35 @@ export class OccurrenceResolver {
     });
   }
 
+  @Query(occurrenceListClassTypeResolver)
+  async FullOccurrenceData(@Args() { selectedIds }: GetFullOccurrenceDataArgs) {
+    return await this.occurrenceService.findOccurrencesByIds(selectedIds);
+  }
+
   @ResolveField('site', siteClassTypeResolver)
   async getSite(@Parent() parent: Occurrence): Promise<Site> {
     return await this.siteService.findOneById(parent.site.id);
   }
 
-  @ResolveField('sample', sampleClassTypeResolver)
-  async getSample(@Parent() parent: Occurrence): Promise<Sample> {
-    return await this.sampleService.findOneById(parent.sample.id);
+  @ResolveField('sample', sampleClassTypeResolver, { nullable: true })
+  async getSample(@Parent() parent: Occurrence): Promise<Sample | null> {
+    if (parent.sample) {
+      return await this.sampleService.findOneById(parent.sample.id);
+    }
+    return Promise.resolve(null);
```

Comment:
I think this can just be `return null;` as the async function handles the resolution of the promise.